lib/provisioner/plugin/utils.rb

```diff
@@ -48,6 +48,16 @@ def to_json(*a)
         end
       end
 
+      # Gets a host's SSH host key
+      def ssh_keyscan(host, type = 'rsa')
+        keytype = type == 'dsa' ? 'dss' : type
+        # TODO: find a way to do this in Ruby
+        key = `ssh-keyscan -t #{type} #{host} 2>&1 | grep #{keytype}`.split(' ')
+        # Bad key type == "unknown key type #{type}"
+        fail "Unknown SSH Key Type: #{type}" if key[2] == 'type'
```

Comment:
add a ``|| key[2].nil?``.  that should also take care of any other concerns about ssh-keyscan failures.